### PR TITLE
[doc][xcluster][2024.1.2] Semi-automatic transactional xCluster

### PR DESCRIPTION
**Pull request by ddhodge**
Database-level setup of transactional xCluster replication

@netlify /preview/deploy/multi-dc/async-replication/async-replication-transactional/

### CODE REVIEW

File: docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+## Set up replication
+
+To set up database-level unidirectional transactional replication, do the following:
+
+1. Create a checkpoint using the following command.
+
+    ```sh
+    ./bin/yb-admin \
+        -master_addresses <primary_master_addresses> \
+        create_xcluster_checkpoint <replication_group_id> <namespace_names>
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md:48]: Lets call it `comma_separated_namespace_names`

File: docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+With database-level setup, xCluster replication operates at the YSQL database granularity. This means you only run xCluster management operations when adding and removing databases from replication, and not when tables in the databases are created or dropped.
+
+In particular, with database-level setup, [DDL changes](#making-ddl-changes) don't require the use of yb-admin. This means DDL changes can be made by any database administrator or user with database permissions, and don't require SSH access or intervention by an IT administrator.
+
+The following assumes you have set up Primary and Standby universes. Refer to [Set up universes](../async-deployment/#set-up-universes).
+
+## Set up replication
+
+To set up database-level unidirectional transactional replication, do the following:
+
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md:42]: We still need to enable PITR the standby database. Same instructions from Manual mode can be copied here

File: docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+1 Universe Replication Groups found: 
+[repl_group1]
+```
+
+## Add a database to a replication group
+
+The database should have at least one table in order to be added to replication. If it is a colocated database then there should be at least one colocated table in the database in order for it to be added to replication.
+
+To add a database to replication, do the following:
+
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md:132]: Enable PITR on standby database

File: docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+You should see output similar to the following:
+
+```output
+Outbound xCluster Replication group rg1 deleted successfully
+```
+
+## Making DDL changes
+
+When performing any DDL operation on databases using database-level transactional xCluster replication (such as creating, altering, or dropping tables or partitions), do the following:
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md:200]:
```suggestion
When performing any DDL operation on databases using database-level transactional xCluster replication (such as creating, altering, or dropping tables, indexes or partitions), do the following:
```

File: docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+1. Execute the DDL on Primary.
+1. Execute the DDL on Standby.
+
+The xCluster configuration is updated automatically. You can insert data into the table as soon as it is created.
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md:205]:
```suggestion
The xCluster configuration is updated automatically. You can insert data into the table as soon as it is created on Primary.
```

**Review by hari90 — CHANGES_REQUESTED** [review]

File: docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup.md [hunk 6]
```diff
@@ -7,11 +7,27 @@ headContent: Set up unidirectional transactional replication
...
-    identifier: async-transactional-setup
+    identifier: async-transactional-setup-1-manual
     weight: 10
+badges: ysql
 type: docs
 ---
 
+<ul class="nav nav-tabs-alt nav-tabs-yb">
+  <li >
+    <a href="../async-transactional-setup/" class="nav-link active">
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup.md:18]: Can we make DB level the default option?

File: docs/content/preview/deploy/multi-dc/async-replication/async-replication-transactional.md [hunk 7]
```diff
@@ -28,13 +29,20 @@ Transactional xCluster support further allows for the role of each universe to s
...
 xCluster safe time is the transactionally consistent time across all tables in a given database at which Reads are served. In the following illustration, T1 is a transactionally consistent time across all tables.
 
 ![Transactional xCluster](/images/deploy/xcluster/xcluster-transactional.png)
 
+## Setup
+
+Transactional xCluster can be set up in the following ways:
+
+- [Fully manual setup](../async-transactional-setup/).
+- [Database-level setup](../async-transactional-setup-dblevel/), providing operationally simpler setup and management of replication, as well as simpler steps for performing DDL changes.
```
[the remark this answers is not in the document]
Comment (hari90) [reply docs/content/preview/deploy/multi-dc/async-replication/async-replication-transactional.md:44]: @lingamsandeep I guess we are calling this GA?

**Comment by hari90** [comment]
All this is available in 2024.1 so you can make the changes to stable as well

File: docs/content/preview/deploy/multi-dc/async-replication/async-replication-transactional.md [hunk 8]
```diff
@@ -28,13 +29,20 @@ Transactional xCluster support further allows for the role of each universe to s
 
 The xCluster role is a property with values ACTIVE or STANDBY that determines and identifies the Primary (source) and Standby (target) universes:
 
-- ACTIVE: The active universe serves both reads & writes. Reads/writes happen as of the latest time and according to the chosen isolation levels.
+- ACTIVE: The active universe serves both reads and writes. Reads/writes happen as of the latest time and according to the chosen isolation levels.
 - STANDBY: The standby universe is meant for reads only. Reads happen as of xCluster safe time for the given database.
```
[the remark this answers is not in the document]
Comment (ddhodge) [reply docs/content/preview/deploy/multi-dc/async-replication/async-replication-transactional.md:34]: @hari90 all of what...

[the remark this answers is not in the document]
Comment (hari90) [reply docs/content/preview/deploy/multi-dc/async-replication/async-replication-transactional.md:34]: Comment moved around.  xCluster role is no longer exists, so you can remove it.

**Comment by ddhodge** [comment]
> All this is available in 2024.1 so you can make the changes to stable as well

Will port changes after getting all comments taken care of

File: docs/content/preview/deploy/multi-dc/async-replication/async-replication-transactional.md [hunk 9]
```diff
@@ -26,15 +27,17 @@ The recovery time objective (RTO) is very low, as it only depends on the applica
...
 xCluster safe time is the transactionally consistent time across all tables in a given database at which Reads are served. In the following illustration, T1 is a transactionally consistent time across all tables.
 
 ![Transactional xCluster](/images/deploy/xcluster/xcluster-transactional.png)
 
+## Setup
+
+Transactional xCluster can be set up in the following ways:
+
+- [Manual setup](../async-transactional-setup/).
+- [Semi-automatic setup](../async-transactional-setup-dblevel/), providing operationally simpler setup and management of replication, as well as simpler steps for performing DDL changes.
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-replication-transactional.md:39]:
```suggestion
- [Manual mode](../async-transactional-setup/).
- [Semi-automatic mode](../async-transactional-setup-dblevel/), providing operationally simpler setup and management of replication, as well as simpler steps for performing DDL changes.
```

File: docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+  </li>
+  <li >
+    <a href="../async-transactional-setup/" class="nav-link">
+      <i class="icon-shell"></i>
+      Manual Setup
+    </a>
+  </li>
+</ul>
+
+Semi-automatic setup of transactional xCluster replication simplifies the operational complexity of managing replication and making DDL changes.
```
Comment (hari90) [inline docs/content/preview/deploy/multi-dc/async-replication/async-transactional-setup-dblevel.md:31]: I see 6 more references to `Semi-automatic setup`. Can we rename them all to `Semi-automatic mode`?

**Review by hari90 — APPROVED** [review]

**Comment by hari90** [comment]
LGTM. Thank you!

**Pull request merged**